File: level/06. Silver 5/1316.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var r = bufio.NewReader(os.Stdin)
var w = bufio.NewWriter(os.Stdout)

func main() {
	defer w.Flush()

	var a, count int
	fmt.Fscanln(r, &a)

	for i := 0; i < a; i++ {
		count++
		z := make(map[string]int)
		var b string
		fmt.Fscanln(r, &b)
		for j:=0;j<len(b);j++{
			if _, t := z[string(b[j])]; t{
				if z[string(b[j])] != j-1{
					count--
					break
				} else{
					z[string(b[j])] = j
				}
			} else{
				z[string(b[j])] = j
			}
		}
	}
	fmt.Fprintln(w,count)
}

File: level/06. Silver 5/1316.py
a = int(input())
count = 0

for i in range(a):
    count += 1
    z = {}
    b = input()
    for j in range(len(b)):
        if b[j] not in z:
            z[b[j]] = j
        else:
            if z[b[j]] != j - 1:
                count -= 1
                break
            else:
                z[b[j]] = j

print(count)
